# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/StaticRange.h
# Commit: 0837e15babab
# Full Hash: 0837e15bababe485cb399b462edcfccd49935126
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2024-03-23 09:29:17
# Regressor Bug: 1867058
# File Overlap Count: 1
# Description:
#   Bug 1867058 - Part 2: Make both StaticRange and nsRange to have a way to cross the trees r=jjaschke,smaug,dom-core
#   
#   For StaticRange, simply allows node to cross the trees.
#   
#   For nsRange, when nodes are in different trees, in addition
# ==============================================================================

diff -r 8ce7972ea4df -r 0837e15babab dom/base/StaticRange.h
--- a/dom/base/StaticRange.h	Fri Mar 22 17:05:48 2024 +0000
+++ b/dom/base/StaticRange.h	Fri Mar 22 17:05:48 2024 +0000
@@ -8,6 +8,7 @@
 #define mozilla_dom_StaticRange_h
 
 #include "mozilla/RangeBoundary.h"
+#include "mozilla/RangeUtils.h"
 #include "mozilla/dom/AbstractRange.h"
 #include "mozilla/dom/StaticRangeBinding.h"
 #include "nsTArray.h"
@@ -70,6 +71,11 @@
    */
   bool IsValid() const;
 
+ private:
+  // Whether the start and end points are in the same tree.
+  // They could be in different trees, i.e, cross shadow boundaries.
+  bool mAreStartAndEndInSameTree = false;
+
  protected:
   explicit StaticRange(nsINode* aNode)
       : AbstractRange(aNode, /* aIsDynamicRange = */ false) {}